### PR TITLE
[mlir][mesh] Remove rank attribute and rename dim_sizes to shape in ClusterOp

### PR DESCRIPTION
**Pull request by sogartar**
Remove the somewhat redundant rank attribute.
Before this change
```
mesh.cluster @mesh(rank = 3, dim_sizes = 2x3)
```
After
```
mesh.cluster @mesh(shape = 2x3x?)
```

The rank is instead determined by the provided shape. With this change no longer `getDimSizes()` can be wrongly assumed to have size equal to the cluster rank.
Now `getShape().size()` will always equal `getRank()`.

### CODE REVIEW

**Comment by sogartar** [comment]
@yaochengji, will you take a look?

**Review by joker-eph — APPROVED** [review]
LGTM. But pleas wait for @yaochengji to have a look

File: mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td [hunk 1]
```diff
@@ -41,7 +41,7 @@ def Mesh_ClusterOp : Mesh_Op<"cluster", [Symbol]> {
     determine the layout and the addressing space of the computation distributed
     across the mesh.
 
-    3. `dim_sizes`: This attribute represents the shape of the device cluster.
+    3. `shape`: This attribute represents the shape of the device cluster.
```
Comment (antiagainst) [inline mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td:44]: We should also remove the doc for `rank` and renumber the rest?

**Review by antiagainst — APPROVED** [review]

**Comment by cxy-1993** [comment]
@joker-eph , are you available? Could help with a code review on this patch.

**Comment by yaochengji** [comment]
LGTM, Thanks.

File: mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td (hunk 1 above)
Comment (sogartar) [reply mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td]: Yes, thank you. I removed it.

**Pull request merged**